### PR TITLE
Replace invariant by warning in createMemoryHistory

### PR DESCRIPTION
**Pull request by petrbrzek**
Related issue #165.

### CODE REVIEW

File: modules/__tests__/MemoryHistory-test.js [hunk 1]
```diff
@@ -53,17 +53,6 @@ describe('memory history', function () {
 
       expect(location.state).toEqual({ id: 5 })
       expect(location.pathname).toEqual('/5')
-
-      expect(function () {
```
Comment (taion) [inline modules/__tests__/MemoryHistory-test.js]: Well, if we already have test coverage, can we just leave the cases there (with some comments) just to check that they don't throw?

File: CHANGES.md [hunk 2]
```diff
@@ -2,15 +2,17 @@
 
 #### Bug fixes
 - Disable browser history on Chrome iOS ([#146])
-- Do not convert same-path PUSH to REPLACE if the hash has changed ([#167]) 
+- Do not convert same-path PUSH to REPLACE if the hash has changed ([#167])
 
 #### Other
 - Add ES2015 module build ([#152])
+- Replace invariant by warning in createMemoryHistory ([#170])
```
Comment (taion) [inline CHANGES.md]: Let's call this a bugfix (:

"Don't throw in memory history when out of history entries"

Comment (mjackson) [inline CHANGES.md]: We can merge and make these kinds of changes ourselves.

Comment (taion) [inline CHANGES.md]: Makes sense - it'd still be good to have the tests to make sure that this doesn't throw, though.

**Comment by taion** [comment]
@petrbrzek Any update?

**Comment by petrbrzek** [comment]
I did some rebase black magic and I'm not sure if I did it correctly. Can you check it please?

**Comment by taion** [comment]
I merged it manually and added a bit more test coverage: https://github.com/rackt/history/commit/e73011a8b50784b293ee890a8ff8534a2ef41769

**Pull request closed without merging**